packages/webapp/pages/join.tsx

```diff
@@ -0,0 +1,199 @@
+            in endless meetings 😜
+          </p>
+          <Button
+            className="max-w-sm h-16 btn-primary !typo-title3"
+            tag="a"
+            href={downloadBrowserExtension}
+            target="_blank"
+            buttonSize={ButtonSize.XLarge}
+          >
+            <FlexCentered className="gap-2">
+              <BrowsersIcon
+                width="50px"
+                height="23px"
+                className="text-theme-label-primary"
+              />
+              Add to browser - it&apos;s free!
+            </FlexCentered>
+          </Button>
+        </FlexCol>
+        <FlexCol className="relative">
+          <Image
+            style={{
+              width: '100%',
+              maxWidth: '1015px',
+              height: '633px',
+              objectFit: 'contain',
+            }}
+            src={cloudinary.referralCampaign.appScreenshot}
+            alt="Image showing the daily.dev app"
+          />
+          <button
+            className="absolute top-1/2 left-1/2 bg-pepper-90 rounded-full transform -translate-x-1/2 -translate-y-1/2"
+            type="button"
+            onClick={() => {
+              setVideoOpen(true);
+            }}
+          >
+            <PlayIcon className="w-28 h-28" size={IconSize.XXXLarge} />
+          </button>
+        </FlexCol>
+      </FlexRow>
+      {isVideoOpen && (
+        <Modal
+          // eslint-disable-next-line @dailydotdev/daily-dev-eslint-rules/no-custom-color
+          className="px-8 bg-[black]"
+          kind={ModalKind.FlexibleCenter}
+          size={ModalSize.XLarge}
+          isOpen={isVideoOpen}
+          onRequestClose={onVideoModalClose}
+        >
+          <CloseButton
+            buttonSize={ButtonSize.Small}
+            className="top-3 right-3 text-white border-white !absolute !btn-secondary"
+            onClick={onVideoModalClose}
+          />
+          <iframe
+            className="w-full border-none aspect-video"
+            // TODO WT-1412-lego-landing-page r.daily.dev link
+            src="https://www.youtube.com/embed/igZCEr3HwCg"
+            title="YouTube video player"
+            allow="encrypted-media;web-share"
+            allowFullScreen
+          />
+        </Modal>
+      )}
+    </div>
+  );
+};
+
+const validateUserId = (value: string) => !!value && value !== '404';
+
+export const getServerSideProps: GetServerSideProps<PageProps> = async ({
+  query,
+  res,
+}) => {
+  const { userid: userId, cid } = query;
+  const campaign = Object.values(ReferralCampaignKey).find(
+    (item) => item === cid,
+  );
+
+  if (!validateUserId(userId as string) || !campaign) {
+    return {
+      redirect: {
+        destination: '/',
+        permanent: false,
+      },
+    };
+  }
+
+  const result = await request<{ user: PageProps['referringUser'] }>(
+    graphqlUrl,
+    GET_REFERRING_USER_QUERY,
+    {
+      id: userId,
+    },
+  );
+
+  if (result) {
+    res.setHeader(
```

Comment:
I think we should set the cookie on the client side because of the gdpr and so on.
We already have a cookie prompt that is shown automatically on first visit

Reply:
I am fine with this, really depending on the legal. Should I then wait for setting a cookie until user click "I like cookies"?

Reply:
Let me open it for discussion on the channel